Client/Frontend/Browser/URLBarView.swift

```diff
@@ -168,7 +165,17 @@ class URLBarView: UIView {
 
         set(newURL) {
             locationView.url = newURL
-            line.isHidden = newURL?.isAboutHomeURL ?? true
+            refreshShieldsStatus()
+        }
+    }
+    
+    /// Update the shields icon based on whether or not shields are enabled for this site
+    func refreshShieldsStatus() {
+        if let domain = currentURL?.normalizedHost, let state = BraveShieldState.getStateForDomain(domain),
+            let shieldsAllOffOverride = state.isShieldOverrideEnabled(.AllOff), shieldsAllOffOverride {
+            shieldsButton.setImage(UIImage(imageLiteralResourceName: "shields-off-menu-icon"), for: .normal)
```

Comment:
Maybe make these images as lazy properties? So they will be assigned only once. I'm not sure if there is a performance benefit though.
`refreshShieldsStatus` is called on every url change btw.

Reply:
I'm fairly certain iOS caches these images once they're loaded once (effectively making them a lazy property) so I don't think this is needed